src/resources.rs

```diff
@@ -223,6 +227,51 @@ pub fn add_tcp_stream(stream: tokio::net::TcpStream) -> Resource {
   Resource { rid }
 }
 
+pub fn add_http_server(s: http_server::HttpServer) -> Resource {
+  let rid = new_rid();
+  let mut tg = RESOURCE_TABLE.lock().unwrap();
+  match tg.insert(rid, Repr::HttpServer(s)) {
+    Some(_) => panic!("There is already a file with that rid"),
+    None => Resource { rid },
+  }
+}
+
+pub fn add_http_transaction(transaction: http_server::Transaction) -> Resource {
+  let rid = new_rid();
+  let mut tg = RESOURCE_TABLE.lock().unwrap();
+  match tg.insert(rid, Repr::HttpTransaction(transaction)) {
+    Some(_) => panic!("There is already a file with that rid"),
+    None => Resource { rid },
+  }
+}
+
+pub fn http_accept(
+  rid: ResourceId,
+) -> impl Future<Item = http_server::Transaction, Error = DenoError> {
+  let mut table = RESOURCE_TABLE.lock().unwrap();
+  let maybe_repr = table.get_mut(&rid);
+  match maybe_repr {
+    Some(Repr::HttpServer(ref mut s)) => Either::A(s.accept()),
+    _ => Either::B(futures::future::err(bad_resource())),
+  }
+}
+
+pub fn http_write_response(
+  rid: ResourceId,
+  response: http_server::Res,
+) -> DenoResult<()> {
+  let mut table = RESOURCE_TABLE.lock().unwrap();
+  let maybe_repr = table.get_mut(&rid);
+  match maybe_repr {
+    Some(Repr::HttpTransaction(ref mut transaction)) => {
+      let tx = transaction.response_tx.take().unwrap();
```

Comment:
So if the user does not ever call `res.writeResponse(...)`, feels like the `HttpTransaction` would stay in resource table forever? (Did I miss something?)

Reply:
That's correct. It's like a socket that doesn't get closed - the fd remains open.

They are removed by the close op
https://github.com/ry/deno/blob/bdcf9e61716c0b02e99b8b29e70aa8b241188189/js/http.ts#L31-L34

Reply:
This `close` seems to be used to close the resource for just the http server, but not the http transactions. (I believe they are 2 separate kind of entities in the resource table)

Reply:
@kevinkassimo I would leave that to the framework layer of abstraction. Docs should be updated with "Caller must writeResponse()"